src/overlays/package_dump.rs

```diff
@@ -0,0 +1,73 @@
+use std::fs::File;
+use std::io::Write;
+use destiny_pkg::TagHash;
+use imgui::Ui;
+use winit::window::Window;
+use crate::overlays::gui::OverlayProvider;
+use crate::packages::package_manager;
+use crate::resources::Resources;
+
+
+pub struct PackageDumper {
+    package_id: String,
+    entry_id: String,
+    message: Result<String, String>
+}
+
+impl PackageDumper {
+
+    pub fn new() -> PackageDumper {
+        return PackageDumper{ package_id: "".to_string(), entry_id: "".to_string(), message: Ok("".to_string()) };
+    }
+
+    fn dump_entry(&self, pkg_id: u16, entry_id: u16) -> bool {
+        let tag = TagHash::new(pkg_id, entry_id);
+        let entry_header = package_manager().get_entry(tag);
+        if entry_header.is_ok() {
+            let entry = entry_header.unwrap();
+            let mut name = format!("{:#010X}", entry.reference);
+            name = name.strip_prefix("0x").unwrap().to_string();
+            let mut file = File::create(format!("{0}.{1}.{2}.tag", name, entry.file_subtype, entry.file_type)).unwrap();
+            if file.write_all(package_manager().read_tag(tag).unwrap().as_slice()).is_err() {
+                println!("Failed to write resource {0} to disk!", format!("{0}.{1}.{2}.tag", name, entry.file_subtype, entry.file_type));
+                return false;
+            } else {
+                return true;
+            }
+        } else {
+            println!("Unable to find resource {0}!", tag);
```

Comment:
Protip, local variables can be embedded in format strings directly
```suggestion
            println!("Unable to find resource {tag}!");
```

Reply:
Note that this only works with variables (eg. `foo`), not with fields (`foo.bar`)